build Failed

#### Problem
<what's wrong or missing, please include any applicable:
 * expected behavior
 * actual behavior
 * steps to reproduce
 * system configuration
 * screenshots, images, logs, etc.
 >

ninja: Entering directory `./out/debug'
[1733/3907] c++ linux_x64_gcc/obj/src/lib/support/tests/libSupportTests.TestStateMachine.cpp.o
FAILED: linux_x64_gcc/obj/src/lib/support/tests/libSupportTests.TestStateMachine.cpp.o
g++ -MMD -MF linux_x64_gcc/obj/src/lib/support/tests/libSupportTests.TestStateMachine.cpp.o.d -Wconversion -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -Wall -Werror -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wno-cast-function-type -fdiagnostics-color -fno-strict-aliasing -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -Wno-implicit-fallthrough -Wno-implicit-fallthrough -std=gnu++14 -fno-rtti -Wnon-virtual-dtor -DCHIP_HAVE_CONFIG_H=1 -I../../src/include -I../../src -Ilinux_x64_gcc/gen/include -I../../zzz_generated/app-common -I../../config/standalone -I../../third_party/nlassert/repo/include -I../../third_party/nlio/repo/include -I../../third_party/nlfaultinjection/repo/include -I../../third_party/inipp/repo/inipp -I../../third_party/nlunit-test/repo/src -I../../third_party/jsoncpp/repo/include -c ../../src/lib/support/tests/TestStateMachine.cpp -o linux_x64_gcc/obj/src/lib/support/tests/libSupportTests.TestStateMachine.cpp.o
../../src/lib/support/tests/TestStateMachine.cpp: In constructor ‘{anonymous}::SimpleStateMachine::SimpleStateMachine()’:
../../src/lib/support/tests/TestStateMachine.cpp:158:41: error: member ‘{anonymous}::SimpleStateMachine::mStateMachine’ is used uninitialized [-Werror=uninitialized]
  158 |     SimpleStateMachine() : mTransitions(mStateMachine), mStateMachine(mTransitions) {}
      |                                         ^~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[1746/3907] c++ linux_x64_gcc/obj/examples/chip-tool/chip-tool.main.cpp.o
ninja: build stopped: subcommand failed.



 #### Proposed Solution
<suggested fix, suggested enhancement>

--- a/src/lib/support/tests/BUILD.gn
+++ b/src/lib/support/tests/BUILD.gn
@@ -62,6 +62,8 @@ chip_test_suite("tests") {
 
   cflags = [
     "-Wconversion",
+
+    # TODO(#21255): work-around for SimpleStateMachine constructor issue.
     "-Wno-error=uninitialized",
   ]
 

--- a/src/lib/support/tests/BUILD.gn
+++ b/src/lib/support/tests/BUILD.gn
@@ -60,7 +60,10 @@ chip_test_suite("tests") {
     test_sources += [ "TestCHIPArgParser.cpp" ]
   }
 
-  cflags = [ "-Wconversion" ]
+  cflags = [
+    "-Wconversion",
+    "-Wno-error=uninitialized",
+  ]
 
   public_deps = [
     "${chip_root}/src/credentials",

